### DISCUSSION TITLE
Re-render specific page

### DISCUSSION THREAD

**Discussion by adeelasd11** [withastro/roadmap#195] [Proposal]
I was wondering if there could be a feature that could allow me to render a single page without rebuilding the entire site. I want to render pages from a CMS. rather than building the entire site when modifying a product I just want to rerender that product page.  

is it possible?

**Comment by delucis** [comment]
This feature is sometimes referred to as supporting "incremental static builds". Astro does not support that at the moment. Depending on your use case you might be able to use SSR (Server-side rendering) instead: in this scenario your server builds each page on demand each time someone wants to visit it.

**Reply by adeelasd11** [reply]
yes but I want to build "on data change" rather than build every time someone visits. Aka I want the advantage that SSG provides without its back draws. Some kind of API that would allow us to build any page whenever we want.

**Reply by delucis** [reply]
Yup, this is what is sometimes called "incremental static builds" and could definitely be explored in the future!

**Comment by adeelasd11** [comment]
looks like NextJs has implemented this https://nextjs.org/blog/next-12-1#on-demand-incremental-static-regeneration-beta
